controllers/auth.js

```diff
@@ -57,33 +57,99 @@ const getUsers = async (req, res = response) => {
 
 
 const deleteUser = async (req, res = response) => {
-    const userId = req.params.id;
-    try {
-      const user = await Event.findById(userId);
-  
-      if (!user) {
-        return res.status(404).json({
-          ok: false,
-          msg: "no user with this id was found",
-        });
-      }
-
-      await User.findByIdAndDelete(userId);
-  
-      res.status(200).json({
-        ok: true,
+  const userId = req.params.id;
+  try {
+    const user = await Event.findById(userId);
+
+    if (!user) {
+      return res.status(404).json({
+        ok: false,
+        msg: "no user with this id was found",
       });
-    } catch (error) {
-      console.log(error);
-      res.status(500).json({
+    }
+
+    await User.findByIdAndDelete(userId);
+
+    res.status(200).json({
+      ok: true,
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      ok: false,
+      msg: "Talk to the administrator",
+    });
+  }
+};
+
+
+const editUser = async (req = request, res = response) => {
+  const userId = req.params.id;
+
+  try {
+    let user = await Event.findById(userId);
+
+    if (!user) {
+      return res.status(404).json({
         ok: false,
-        msg: "Talk to the administrator",
+        msg: "No user with this id was found",
       });
     }
-  };
+
+    const newUser = {
+      ...req.body,
+    };
+
+    const updatedUser = await User.findByIdAndUpdate(userId, newUser, { new: true });
+
+    res.status(200).json({
+      ok: true,
+      user: updatedUser,
+    });
+
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      ok: false,
+      msg: "Error searching for the user in the database, talk to an administrator.",
+    });
+  }
+};
+
+
+const loginUser = async (req, res = response) => {
+  const { email, password } = req.body;
+
+  try {
+
+    let user = await User.findOne({ email });
+
+    if (!user || !bcrypt.compareSync(password, user.password)) {
+      return res.status(400).json({
+        ok: false,
+        msg: "Incorrect email or password",
+      });
+    };
+
+    return res.status(200).json({
```

Comment:
the user's status must be sent to know if it is the first time he/she enters the system